src/densenet.jl

```diff
@@ -1,92 +1,57 @@
-struct Bottleneck
-    layer
-end
-
-@functor Bottleneck
+using Flux
 
-Bottleneck(in_planes, growth_rate) = Bottleneck(Chain(BatchNorm(in_planes, relu),
-                                          Conv((1, 1), in_planes => 4growth_rate),
-                                          BatchNorm(4growth_rate, relu),
-                                          Conv((3, 3), 4growth_rate => growth_rate, pad = (1, 1))))
+function bottleneck(inplanes, growth_rate)
+    inner_channels = 4 * growth_rate
+    m = Chain(conv_bn((1, 1), inplanes, inner_channels; usebias=false, rev=true)...,
+              conv_bn((3, 3), inner_channels, growth_rate; pad=1, usebias=false, rev=true)...)
 
-(b::Bottleneck)(x) = cat(b.layer(x), x, dims = 3)
+    layer = x -> begin
+        y = m(x)
+        return cat(x, y; dims=3)
+    end
+end
 
-Transition(chs::Pair{<:Int,<:Int}) = Chain(BatchNorm(chs[1], relu),
-                                            Conv((1, 1), chs),
-                                            MeanPool((2, 2)))
+transition(inplanes, outplanes) = Chain(conv_bn((1, 1), inplanes, outplanes; usebias=false, rev=true)...,
+                                        MeanPool((2, 2)))
 
-function _make_dense_layers(block, in_planes, growth_rate, nblock)
-    local layers = []
+function make_dense_block(inplanes, growth_rate, nblock)
+    layers = []
     for i in 1:nblock
-        push!(layers, block(in_planes, growth_rate))
-        in_planes += growth_rate
+        push!(layers, bottleneck(inplanes, growth_rate))
+        inplanes += growth_rate
     end
     Chain(layers...)
```

Comment:
By convention, we haven't been returning intermediate `Chain`s. Here you can just return `layers`.